### PR TITLE
MTM-49761 added Microservice Security troubleshooting

### PR DESCRIPTION
**Pull request by kaczynskid-sag**

### CODE REVIEW

**Review by pawelkowalczyksag — APPROVED** [review]

File: content/microservice-sdk/java-bundle/troubleshooting.md [hunk 1]
```diff
@@ -122,3 +122,12 @@ In order to fix this, do the following:
...
 Also refer to [Docker Engine > Installation per distro > Optional post-installation steps](https://docs.docker.com/engine/install/linux-postinstall/) in the Docker documentation.
+
+##### Customizing Microservice Security configuration
+
+Starting from version 10.17 there was a change in internal Microservice Security configuration following
+deprecation of `WebSecurityConfigurerAdapter` by Spring Security. Microservice SDK is not using direct
+declaration of `SecurityFilterChain` bean in its internal configuration any more. At the same time Spring Security
+only allows one of those configuration approaches in a single application. This means that if the old,
+adapter based method was used in your code before, you will have to migrate to the new, direct filters
+declaration for application to start. Please refer to [Spring Security documentation](https://docs.spring.io/spring-security/reference/5.8/migration/servlet/config.html#_stop_using_websecurityconfigureradapter) for more details.
```
Comment (jkot-softwareag) [inline content/microservice-sdk/java-bundle/troubleshooting.md:133]: Do we have something in our examples? If so I would add in last sentence:
`Please refer to [Spring Security documentation] and examples in [cumulocity-examples] respository for more details.

**Review by jkot-softwareag — APPROVED** [review]

File: content/microservice-sdk/java-bundle/troubleshooting.md (hunk 1 above)
Comment (kaczynskid-sag) [reply content/microservice-sdk/java-bundle/troubleshooting.md:133]: No, we do not have any custom security configuration in examples.

Comment (jkot-softwareag) [reply content/microservice-sdk/java-bundle/troubleshooting.md:133]: :-(  maybe we should have at least one...

File: content/microservice-sdk/java-bundle/troubleshooting.md [hunk 2]
```diff
@@ -122,3 +122,12 @@ In order to fix this, do the following:
 4. Try running a Docker command again.
 
 Also refer to [Docker Engine > Installation per distro > Optional post-installation steps](https://docs.docker.com/engine/install/linux-postinstall/) in the Docker documentation.
+
+##### Customizing Microservice Security configuration
```
Comment (BeateRixen) [inline content/microservice-sdk/java-bundle/troubleshooting.md:126]:
```suggestion
##### Customizing microservice security configuration
```

File: content/microservice-sdk/java-bundle/troubleshooting.md [hunk 3]
```diff
@@ -122,3 +122,12 @@ In order to fix this, do the following:
...
+
+Starting from version 10.17 there was a change in internal Microservice Security configuration following
```
Comment (BeateRixen) [inline content/microservice-sdk/java-bundle/troubleshooting.md:128]:
```suggestion
Starting from version 10.17 there was a change in the internal microservice security configuration following
```

File: content/microservice-sdk/java-bundle/troubleshooting.md [hunk 4]
```diff
@@ -122,3 +122,12 @@ In order to fix this, do the following:
...
+deprecation of `WebSecurityConfigurerAdapter` by Spring Security. Microservice SDK is not using direct
```
Comment (BeateRixen) [inline content/microservice-sdk/java-bundle/troubleshooting.md:129]:
```suggestion
the deprecation of `WebSecurityConfigurerAdapter` by Spring Security. The Microservice SDK does no longer use a direct
```

File: content/microservice-sdk/java-bundle/troubleshooting.md [hunk 5]
```diff
@@ -122,3 +122,12 @@ In order to fix this, do the following:
...
+declaration of `SecurityFilterChain` bean in its internal configuration any more. At the same time Spring Security
```
Comment (BeateRixen) [inline content/microservice-sdk/java-bundle/troubleshooting.md:130]:
```suggestion
declaration of the `SecurityFilterChain` bean in its internal configuration. At the same time, Spring Security
```

File: content/microservice-sdk/java-bundle/troubleshooting.md [hunk 6]
```diff
@@ -122,3 +122,12 @@ In order to fix this, do the following:
...
+only allows one of those configuration approaches in a single application. This means that if the old,
```
Comment (BeateRixen) [inline content/microservice-sdk/java-bundle/troubleshooting.md:131]:
```suggestion
only allows one of these configuration approaches in a single application. This means that if the old,
```

File: content/microservice-sdk/java-bundle/troubleshooting.md [hunk 7]
```diff
@@ -122,3 +122,12 @@ In order to fix this, do the following:
...
+adapter based method was used in your code before, you will have to migrate to the new, direct filters
```
Comment (BeateRixen) [inline content/microservice-sdk/java-bundle/troubleshooting.md:132]:
```suggestion
adapter-based method has been used in your code before, you will have to migrate to the new, direct filters
```

File: content/microservice-sdk/java-bundle/troubleshooting.md (hunk 1 above)
Comment (BeateRixen) [inline content/microservice-sdk/java-bundle/troubleshooting.md:133]:
```suggestion
declaration for applications to start. Refer to the [Spring Security documentation](https://docs.spring.io/spring-security/reference/5.8/migration/servlet/config.html#_stop_using_websecurityconfigureradapter) for more details.
```

**Comment by BeateRixen** [comment]
@kaczynskid-sag This should probably better go into the announcements section of the release notes than into the user documentation?

**Comment by kaczynskid-sag** [comment]
> @kaczynskid-sag This should probably better go into the announcements section of the release notes than into the user documentation?

@BeateRixen I added this info to release notes of the source task https://cumulocity.atlassian.net/browse/MTM-49761. This is an additional troubleshooting section in case some misses the release notes or just starts using SDK after this change and encounters described problems. I was advised by @jkot-softwareag to add this on top of release notes to possibly decrease the number of possible support requests or to make them easier to handle by just providing a link to documentation.

**Comment by BeateRixen** [comment]
> > @kaczynskid-sag This should probably better go into the announcements section of the release notes than into the user documentation?
> 
> @BeateRixen I added this info to release notes of the source task https://cumulocity.atlassian.net/browse/MTM-49761. This is an additional troubleshooting section in case some misses the release notes or just starts using SDK after this change and encounters described problems. I was advised by @jkot-softwareag to add this on top of release notes to possibly decrease the number of possible support requests or to make them easier to handle by just providing a link to documentation.

@kaczynskid-sag This is the standard release notes entry. I would better add it to the SDK section in the announcements section for 10.17, like here (10.16 example): https://staging-resources.cumulocity.com/releasenotes/release-10-16-0/announcements-10-16-0/. If we add such information to the docs, someone must take care of removing it in some version again, otherwise it will stay there forever. What do you think @jkot-softwareag ?

**Comment by jkot-softwareag** [comment]
> This is the standard release notes entry. I would better add it to the SDK section in the announcements section for 10.17, like here (10.16 example): https://staging-resources.cumulocity.com/releasenotes/release-10-16-0/announcements-10-16-0/. If we add such information to the docs, someone must take care of removing it in some version again, otherwise it will stay there forever. What do you think @jkot-softwareag ?

If we check https://cumulocity.com/guides/microservice-sdk/java/#developing-microservice we have following entry (see screenshot), in a bit different place (above the current). I was thinking to add similar for upgrading to next SDK version and next Spring-Boot version. Especially when teams like Apama noticed change after their microservice stopped to build properly. There can be multiple clients like this up there, and it is easier to redirect them to documentation than to release notes. Documentation will only state about migrating to particular version of java SKD and particular spring-boot version, which will not change after release.

image

**Comment by BeateRixen** [comment]
> > This is the standard release notes entry. I would better add it to the SDK section in the announcements section for 10.17, like here (10.16 example): https://staging-resources.cumulocity.com/releasenotes/release-10-16-0/announcements-10-16-0/. If we add such information to the docs, someone must take care of removing it in some version again, otherwise it will stay there forever. What do you think @jkot-softwareag ?
> 
> If we check https://cumulocity.com/guides/microservice-sdk/java/#developing-microservice we have following entry (see screenshot), in a bit different place (above the current). I was thinking to add similar for upgrading to next SDK version and next Spring-Boot version. Especially when teams like Apama noticed change after their microservice stopped to build properly. There can be multiple clients like this up there, and it is easier to redirect them to documentation than to release notes. Documentation will only state about migrating to particular version of java SKD and particular spring-boot version, which will not change after release.
> 
> image
> 
> Maybe we should move Dariusz entry above, to have it in same chapter/section as "upgrading to 10.13".

Yes, sounds reasonable to me. I was concerned because it says "Starting from 10.17...". Such an information is not really relevant after some time, and usually only provided in the release announcements.

**Comment by kaczynskid-sag** [comment]
> Yes, sounds reasonable to me. I was concerned because it says "Starting from 10.17...". Such an information is not really relevant after some time, and usually only provided in the release announcements.

@BeateRixen moved to https://cumulocity.com/guides/microservice-sdk/java/#developing-microservice

**Review by BeateRixen — APPROVED** [review]

**Pull request merged**